Meme/Meme/NetworkClient.swift

```diff
@@ -12,20 +12,11 @@ class NetworkClient {
 
     fileprivate func getTemplates(_ data: Data?) -> [Template]? {
         guard let data = data,
-            let jsonResponse = try? JSONSerialization.jsonObject(with: data, options: []) as? [String:Any] else {
+            let jsonResponse = try? JSONSerialization.jsonObject(with: data, options: []) as? [String:String] else {
                 print("Error deserializing template data");
                 return nil
         }
-
-        var templates = [Template]()
-        jsonResponse.forEach { (key, value) in
-            var dictionary = [String:Any]()
-            dictionary[key] = value
-            if let template = Template.init(dictionary: dictionary) {
-                templates.append(template)
-            }
-        }
-        return templates
+        return jsonResponse.compactMap({ return Template.init(name: $0.0, templateURL:URL(string: $0.1)!)})
```

Comment:
@vhart Any advice on how to avoid the `!` and also use compactMap here. Issue is that the Template struct initializes with a URL which is fallible when getting constructed with string.

Reply:
@mesbekmek Template should take in a string url and handle the parsing in the initializer.